### PR TITLE
Adding a "pubspec for Flutter" page

### PR DESCRIPTION
**Pull request by sfshaza2**
Staged:  https://sz-flutter.firebaseapp.com/docs/development/tools/pubspec

### CODE REVIEW

**Review by bwilkerson — APPROVED** [review]
This looks very good to me! It's probably worth having someone on the Flutter team that's more familiar with this also look at it.

There are a couple of items (at the level of 'assets') that I found that aren't explicitly listed here. They might well be described in some of the documents you pointed to. It's also possible that they're obsolete or not worth documenting, but I'm listing them in case you want to explore them further:
- 'generate'
- 'module'
- 'plugin'

File: src/docs/development/tools/pubspec.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
+---
+title: "Flutter and the pubspec file"
+description: "Describes the Flutter-only fields in the pubspec file."
+---
+
+Every Flutter project includes a `pubspec.yaml` file,
```
Comment (jmagman) [inline src/docs/development/tools/pubspec.md:6]: Maybe a mention that it's written in YAML, with a link to the somewhat inaccessible documentation https://yaml.org/spec/1.2/spec.html

File: src/docs/development/tools/pubspec.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+often referred to as _the pubspec_.
+A basic pubspec is generated when you create
+a new Flutter project. It's located at the top
+of the project tree and contains metadata about
+the project that the Dart and Flutter tooling
+needs to know.
+
+The pubspec file specifies dependencies
+that the project requires, such as particular packages
```
Comment (jmagman) [inline src/docs/development/tools/pubspec.md:15]: This page seems to mostly be relevant to apps, though there's some overlap.  Maybe move the `/docs/development/packages-and-plugins/developing-packages` link up to point out where to look for more info if they are writing packages and plugins?

**Review by jmagman — COMMENTED** [review]
Some minor nits, LGTM

File: src/docs/development/tools/pubspec.md (hunk 2 above)
Comment (sfshaza2) [reply src/docs/development/tools/pubspec.md:27]: Great idea. Done!

**Comment by sfshaza2** [comment]
Thanks, @bwilkerson and @jmagman for your reviews!

**Pull request merged**